### PR TITLE
Add Columbus day to InternationalHolidays

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

This PR adds `_add_columbus_day(...)` for Latin American countries which are celebrated on October 12 annually under various names such as "Indigeneous Resistance Day", "Respect for Cultural Diversity Day", and "Day of the Race".

Other changes introduced along with this PR:
- Migrate Honduras holidays to holiday groups
- Apply `_add_columbus_day(...)` for `HN`, `VE`

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)x
## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/venezuela.py [hunk 1]
```diff
@@ -85,10 +85,10 @@ def _populate(self, year):
 
         if year >= 2002:
             # Day of Indigenous Resistance.
-            self._add_holiday(tr("Día de la Resistencia Indígena"), OCT, 12)
+            self._add_columbus_day(tr("Día de la Resistencia Indígena"))
         elif year >= 1921:
             # Columbus Day.
-            self._add_holiday(tr("Día de la Raza"), OCT, 12)
+            self._add_columbus_day(tr("Día de la Raza"))
```
Comment (KJhellico) [inline holidays/countries/venezuela.py:91]:
```python
        if year >= 1921:
            self._add_columbus_day(
                # Day of Indigenous Resistance.
                tr("Día de la Resistencia Indígena")
                if year >= 2002
                # Columbus Day.
                else tr("Día de la Raza")
            )
```

**Comment by arkid15r** [comment]
I recommend splitting this PR into at least 2 for better review experience:

```
<!--
  Type of change you want to introduce. Please, check one (1) box only!
  If your PR requires multiple boxes to be checked, most likely it needs to
  be split into multiple PRs.
-->
```
  - [x] Supported country/market holidays update (calendar discrepancy fix, localization)
 -  [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)

**Comment by PPsyrius** [comment]
@arkid15r @KJhellico I'll keep this one for the new Columbus Day addition to `InternationalHolidays` only and migrate the subsequent national-level changes to their own respective drafted PR then.

File: holidays/countries/venezuela.py (hunk 1 above)
Comment (PPsyrius) [reply holidays/countries/venezuela.py]: Fixed.

**Comment by PPsyrius** [comment]
This PR is now refactored to only include the basic `_add_columbus_day(...)` for `HN` and `VE`, with subsequent logic changes split into their own PRs. @arkid15r @KJhellico

**Comment by arkid15r** [comment]
> This PR is now refactored to only include the basic `_add_columbus_day(...)` for `HN` and `VE`, with subsequent logic changes split into their own PRs. @arkid15r @KJhellico

Ideally, it'd include [L64](https://github.com/dr-prodigy/python-holidays/pull/1256/files#diff-51871a4040f5aa28c85c10ff8f7f6dd47fe10d27f9b8863a80ba0b2bb06de672R64) related changes only for HN but that's fine.

Just a minor suggestion to move the method up to keep the current ordering.

**Comment by PPsyrius** [comment]
I've just noticed that holiday group entries are supposed to be ordered alphabetically, fixed now 🙏 @arkid15r

**Review by arkid15r — COMMENTED** [review]
LGTM 👍

**Review by KJhellico — COMMENTED** [review]
All looks good, but add these missing strings, please (I missed them when creating the translation 😕), and make the appropriate changes to the tests.

**Comment by KJhellico** [comment]
Thank you! Now everything is fine 👍

**Comment by KJhellico** [comment]
It has been merged into beta. Free way for Argentina and Colombia updates 👍

**Pull request merged**